src/scripts/main.js

```diff
@@ -1 +1,80 @@
 'use strict';
+
+document.addEventListener('contextmenu', (e) => e.preventDefault());
+
+const showNotification = (message, type) => {
+  document.body.insertAdjacentHTML(
+    'afterbegin',
+    `<div class="${type}" data-qa="notification">${message}</div>`,
+  );
+};
+
+const success = (message) => showNotification(message, 'success');
+const error = (message) => showNotification(message, 'error');
+
+const firstPromise = new Promise((resolve, reject) => {
+  const clickHandler = (e) => {
+    if (e.button === 0) {
+      resolve();
+      document.removeEventListener('click', clickHandler);
+    }
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
```

Comment:
The error message 'First promise was rejected' is hardcoded. Consider making this message more dynamic or configurable if needed.